basic_pipeline/11_Pipeline.md

```diff
@@ -45,110 +45,57 @@ With such a concept in mind, it's possible to create reliable and fault-tolerant
-Of course, there is also a `via_out/1` function, which is used to point the output pad with the given identifier, but there was no need to use it.
-In the case of other elements we do not need to explicitly point the desired pads since we are taking advantage of the default pads name - `:input` for the input pads and `:output` for the output ones (see what names we have given to our pads in the elements other than the mixer!). However, we could rewrite the following `links` definitions and explicitly specify the pad names:
-
-**_`lib/Pipeline.ex`_**
-
-```elixir
-
-def handle_init(_opts) do
+defmodule Basic.Pipeline do
  ...
- links = [
-  link(:input1) |> via_out(:output) |> via_in(:input) |> to(:ordering_buffer1) |> via_out(:output) |> via_in(:input) |> to(:depayloader1),
-  link(:input2) |> via_out(:output) |> via_in(:input) |> to(:ordering_buffer2) |> via_out(:output) |> via_in(:input) |>to(:depayloader2),
-  link(:depayloader1) |> via_out(:output) |> via_in(:first_input) |> to(:mixer),
-  link(:depayloader2) |> via_out(:output) |> via_in(:second_input) |> to(:mixer),
-  link(:mixer) |> via_out(:output) |> via_in(:input) |> to(:output)
- ]
+  @impl true
+  def handle_init(_context, _options) do
+    spec = [
+      child(:input1, %Basic.Elements.Source{location: "input.A.txt"})
+      |> child(:ordering_buffer1, Basic.Elements.OrderingBuffer)
+      |> child(:depayloader1, %Basic.Elements.Depayloader{packets_per_frame: 4})
+      |> via_in(:first_input)
+      |> child(:mixer, Basic.Elements.Mixer)
+      |> child(:output, %Basic.Elements.Sink{location: "output.txt"}),
+      child(:input2, %Basic.Elements.Source{location: "input.B.txt"})
+      |> child(:ordering_buffer2, Basic.Elements.OrderingBuffer)
+      |> child(:depayloader2, %Basic.Elements.Depayloader{packets_per_frame: 4})
+      |> via_in(:second_input)
+      |> get_child(:mixer)
+    ]
+
+    {[spec: spec], %{}}
+  end
  ...
-end
 ```
 
-That's almost it! All we need to do is to return a proper tuple from the `handle_init/1` callback, with the `:spec` action defined:
-
-**_`lib/Pipeline.ex`_**
-
-```elixir
+Remember to pass the desired file paths in the `:location` option! 
 
-def handle_init(_opts) do
- ...
- spec = %ParentSpec{children: children, links: links}
- { {:ok, spec: spec}, %{} }
-end
-```
+Now... that's it! :) 
+The spec list using Membrane's DSL is enough to describe our pipeline's topology. The child keywords spawn components of the specified type and we can use the `|>` operator to link them together. When the pads that should be linked are unamibiguous this is straightforward but for links like those with `Mixer` we can specify the pads using `via_in/1`. There also exists a `via_out/1` keyword which works in a similar way. 
+As you can see the first argument to `child/2` is a component identifier, but it's also possible to have anonymous children using `child/1`, which just has Membrane generate a unique id under the hood.
```

Comment:
Perhaps we could link to the documentation for these `child/2` etc.? In fact, there is `child/1`, `child/2`, `child/3`, and `child/4`, depending on the place where the child is put in the pipeline (for children at the beggining of the pipeline the arity of function is always one less :D )